app/actions/views/user_profile.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {getDirectChannelName} from 'service/utils/channel_utils';
+import {createDirectChannel} from 'service/actions/channels';
+import {getTeamMember} from 'service/actions/teams';
+import {handleSelectChannel, toggleDMChannel} from 'app/actions/views/channel';
+import {NavigationTypes} from 'app/constants';
+
+export function makeDirectChannel(otherUserId) {
```

Comment:
@jarredwitt there's already an action for this take a look at https://github.com/mattermost/mattermost-mobile/blob/master/app/actions/views/more_dms.js#L9